kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/AiWorkflow.kt

```diff
@@ -78,7 +78,10 @@ class AiWorkflow(
         LEFT -> UP
       }
       println("AI changing direction from ${state.direction} to $newDirection")
-      return@runningWorker enterState(state.copy(direction = newDirection))
+      return@runningWorker WorkflowAction {
```

Comment:
Probably better to pull this whole block out into a `handleDirectionChange` method, but I'd still write that method using `WorkflowAction {`.